airflow/providers/amazon/aws/operators/emr_containers.py

```diff
@@ -0,0 +1,151 @@
+        specifically either application configuration or monitoring configuration.
+    :type configuration_overrides: dict
+    :param client_request_token: The client idempotency token of the job run request.
+        Use this if you want to specify a unique ID to prevent two jobs from getting started.
+        If no token is provided, a UUIDv4 token will be generated for you.
+    :type client_request_token: str
+    :param aws_conn_id: The Airflow connection used for AWS credentials.
+    :type aws_conn_id: str
+    :param sleep_time: Time (in seconds) to wait between two consecutive calls to check query status on EMR
+    :type sleep_time: int
+    :param max_tries: Maximum number of times to wait for the job run to finish.
+        Defaults to None, which will poll until the job is *not* in a pending, submitted, or running state.
+    :type max_tries: int
+    """
+
+    template_fields = ["name", "virtual_cluster_id", "execution_role_arn", "release_label", "job_driver"]
+    ui_color = "#f9c915"
+
+    @apply_defaults
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        *,
+        name: str,
+        virtual_cluster_id: str,
+        execution_role_arn: str,
+        release_label: str,
+        job_driver: dict,
+        configuration_overrides: Optional[dict] = None,
+        client_request_token: Optional[str] = None,
+        aws_conn_id: str = "aws_default",
+        sleep_time: int = 30,
+        max_tries: Optional[int] = None,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.name = name
+        self.virtual_cluster_id = virtual_cluster_id
+        self.execution_role_arn = execution_role_arn
+        self.release_label = release_label
+        self.job_driver = job_driver
+        self.configuration_overrides = configuration_overrides or {}
+        self.aws_conn_id = aws_conn_id
+        self.client_request_token = client_request_token or str(uuid4())
+        self.sleep_time = sleep_time
+        self.max_tries = max_tries
+        self.job_id = None
+
+    @cached_property
+    def hook(self) -> EMRContainerHook:
+        """Create and return an EMRContainerHook."""
+        return EMRContainerHook(
+            self.aws_conn_id,
+            virtual_cluster_id=self.virtual_cluster_id,
+        )
+
+    def execute(self, context: dict) -> Optional[str]:
+        """Run job on EMR Containers"""
+        self.job_id = self.hook.submit_job(
+            self.name,
+            self.execution_role_arn,
+            self.release_label,
+            self.job_driver,
+            self.configuration_overrides,
+            self.client_request_token,
+        )
+        query_status = self.hook.poll_query_status(self.job_id, self.max_tries, self.sleep_time)
+
+        if query_status in EMRContainerHook.FAILURE_STATES:
+            # self.hook.get_state_change_reason(self.query_execution_id)
+            error_message = "BEEP BOOP"
```

Comment:
More useful error message, perhaps? :)

Reply:
Updated to retrieve failure reason from the job description. :)